src/mpl/include/mpl_atomic.h

```diff
@@ -107,6 +107,10 @@ static void MPL_atomic_compiler_barrier(void);
 #elif defined(MPL_HAVE_NT_INTRINSICS)
 #include "mpl_atomic_nt_intrinsics.h"
 #elif defined(MPL_USE_LOCK_BASED_PRIMITIVES)
+#include <pthread.h>
+typedef pthread_mutex_t MPL_emulation_ipl_t;
```

Comment:
I'll need look it every time for `MPL_emulation_ipl_t` :) How about a more intuitive name, e.g. `MPL_atomic_lock_t`?

Reply:
`MPL_atomic_lock_t` is confusing since it is not a lock for ordinary "lock" use (such as spinlock and mutex). It is an internal lock for  `MPL_USE_LOCK_BASED_PRIMITIVES` and ideally should not be revealed, though it is needed since its size is necessary to allocate memory. Actually, I was confused when I looked at this for the first time

However, this name is not intuitive. In addition, `MPL_atomic` must be a proper prefix for this. Any name is okay if it is clear that this lock is not for users but for `MPL_USE_LOCK_BASED_PRIMITIVES`. How about one of the following?
- `MPL_atomic_internal_interprocess_lock_t`
- `MPL_atomic_internal_lock_t`
- `MPL_atomic_emulation_ipl_t`
- `MPL_atomic_internal_mutex_t`
- `MPL_atomic_internal_ipl_t`

Reply:
I pick MPL_atomic_internal_lock_t

Reply:
OK. I will rename this type. Thanks!